odoo/tools/translate.py

```diff
@@ -297,7 +297,10 @@ def process(node):
 
 
 def parse_xml(text):
-    return etree.fromstring(text)
+    try:
+        return etree.fromstring(text)
+    except ValueError as e:
+        raise UserError(str(e))
```

Comment:
```suggestion
    if isinstance(text, str):
        if re.search(r'<\?xml[^>]*encoding=.*?\?>', text, re.IGNORECASE):
            raise UserError(_(
                "Unicode strings with encoding declaration are not supported in XML. "
                "Remove the encoding declaration."
            ))
    return etree.fromstring(text)
```
IMO, it is good we raise an error when the Unicode declaration is not found in XML by re.search. 


Also handle same issue below while creating type without name https://github.com/odoo/odoo/blob/80ff2d2e16e5ee7618d69e5a418a955cd2e80cfc/odoo/addons/base/models/ir_ui_view.py#L485